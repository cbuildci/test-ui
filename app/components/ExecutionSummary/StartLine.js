/**
 * StartLine
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative } from 'react-intl';
import messages from './messages';
import Time from 'components/Time';

function StartLine({
    githubHost,
    owner,
    repo,
    createTime,
    event,
}) {
    let message = messages.startedWithoutEvent;

    const values = {
        relativeCreateTimeHtml: (
            <strong>
                <Time date={createTime}>
                    <FormattedRelative
                        value={createTime}
                        updateInterval={5000}
                    />
                </Time>
            </strong>
        ),
    };

    if (event) {
        message = messages.startedWithUnknownEvent;
        values.event = event.event;

        if (event.event === 'pull_request') {
            message = messages.startedForPullRequest;

            values.action = event.action;
            values.actionHtml = (
                <strong>
                    <FormattedMessage
                        {...messages.startedForPullRequest_action}
                        values={{ action: event.action }}
                    />
                </strong>
            );

            values.pullRequestLinkHtml = (
                <a href={`https://${githubHost}/${owner}/${repo}/pull/${event.pull_request.number}`}>
                    <FormattedMessage
                        {...messages.startedForPullRequest_link}
                        values={{ pullRequestNumber: event.pull_request.number }}
                    />
                </a>
            );
        }
        else if (event.event === 'check_run'
            && event.action === 'requested_action'
            && event.requested_action.identifier === 'rerun') {

            message = messages.startedByUserAction;

            values.username = event.sender.login;
            values.usernameHtml = (
                <a href={`https://${githubHost}/${event.sender.login}`}>@{event.sender.login}</a>
            );

            values.identifier = event.requested_action.identifier;
            values.identifierHtml = (
                <strong>
                    <FormattedMessage
                        {...messages.startedByUserAction_identifier}
                        values={{ identifier: event.requested_action.identifier }}
                    />
                </strong>
            );
        }
    }

    return (
        <FormattedMessage
            {...message}
            values={values}
        />
    );
}

StartLine.propTypes = {
    githubHost: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    createTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]).isRequired,
    conclusionTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
    event: PropTypes.object,
};

StartLine.defaultProps = {
    conclusionTime: null,
    event: null,
};

export default StartLine;
