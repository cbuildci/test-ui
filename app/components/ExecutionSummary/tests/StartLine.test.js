import React from 'react';
import { shallow, render } from 'enzyme';
import { IntlProvider } from 'react-intl';
import StartLine from '../StartLine';

describe('<StartLine/>', () => {
    it('should render expected JSX with just the create time', () => {
        const jsx = (
            <StartLine
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
            />
        );

        expect(shallow(jsx)).toMatchSnapshot();
        expect(render(
            <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
        )).toMatchSnapshot();
    });

    it('should render expected JSX for "opened" pull request event', () => {
        const jsx = (
            <StartLine
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'pull_request',
                    action: 'opened',
                    pull_request: {
                        number: 5,
                    },
                }}
            />
        );

        expect(shallow(jsx)).toMatchSnapshot();
        expect(render(
            <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
        )).toMatchSnapshot();
    });

    it('should render expected JSX for "synchronize" pull request event', () => {
        const jsx = (
            <StartLine
                githubHost="git.com"
                owner="foo"
                repo="bar"
                createTime={1500000000000}
                event={{
                    event: 'pull_request',
                    action: 'synchronize',
                    pull_request: {
                        number: 5,
                    },
                }}
            />
        );

        expect(shallow(jsx)).toMatchSnapshot();
        expect(render(
            <IntlProvider locale="en" initialNow={1500000100000}>{jsx}</IntlProvider>
        )).toMatchSnapshot();
    });
});
