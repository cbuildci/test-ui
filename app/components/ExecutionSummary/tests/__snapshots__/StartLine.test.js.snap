// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartLine/> should render expected JSX for "opened" pull request event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml}"
  id="app.components.ExecutionSummary.startedForPullRequest"
  values={
    Object {
      "action": "opened",
      "actionHtml": <strong>
        <FormattedMessage
          defaultMessage="{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }"
          id="app.components.ExecutionSummary.startedForPullRequest_action"
          values={
            Object {
              "action": "opened",
            }
          }
        />
      </strong>,
      "event": "pull_request",
      "pullRequestLinkHtml": <a
        href="https://git.com/foo/bar/pull/5"
      >
        <FormattedMessage
          defaultMessage="pull request #{pullRequestNumber}"
          id="app.components.ExecutionSummary.startedForPullRequest_link"
          values={
            Object {
              "pullRequestNumber": 5,
            }
          }
        />
      </a>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<StartLine/> should render expected JSX for "opened" pull request event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        2 minutes ago
      </span>
    </time>
  </strong>
   after 
  <strong>
    <span>
      opening
    </span>
  </strong>
  
        
  <a
    href="https://git.com/foo/bar/pull/5"
  >
    <span>
      pull request #5
    </span>
  </a>
</span>
`;

exports[`<StartLine/> should render expected JSX for "synchronize" pull request event 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml}"
  id="app.components.ExecutionSummary.startedForPullRequest"
  values={
    Object {
      "action": "synchronize",
      "actionHtml": <strong>
        <FormattedMessage
          defaultMessage="{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }"
          id="app.components.ExecutionSummary.startedForPullRequest_action"
          values={
            Object {
              "action": "synchronize",
            }
          }
        />
      </strong>,
      "event": "pull_request",
      "pullRequestLinkHtml": <a
        href="https://git.com/foo/bar/pull/5"
      >
        <FormattedMessage
          defaultMessage="pull request #{pullRequestNumber}"
          id="app.components.ExecutionSummary.startedForPullRequest_link"
          values={
            Object {
              "pullRequestNumber": 5,
            }
          }
        />
      </a>,
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<StartLine/> should render expected JSX for "synchronize" pull request event 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        2 minutes ago
      </span>
    </time>
  </strong>
   after 
  <strong>
    <span>
      push
    </span>
  </strong>
   to
        
  <a
    href="https://git.com/foo/bar/pull/5"
  >
    <span>
      pull request #5
    </span>
  </a>
</span>
`;

exports[`<StartLine/> should render expected JSX with just the create time 1`] = `
<FormattedMessage
  defaultMessage="Started {relativeCreateTimeHtml}"
  id="app.components.ExecutionSummary.startedWithoutEvent"
  values={
    Object {
      "relativeCreateTimeHtml": <strong>
        <InjectIntl(Time)
          date={1500000000000}
        >
          <FormattedRelative
            updateInterval={5000}
            value={1500000000000}
          />
        </InjectIntl(Time)>
      </strong>,
    }
  }
/>
`;

exports[`<StartLine/> should render expected JSX with just the create time 2`] = `
<span>
  Started 
  <strong>
    <time
      class="position-relative"
      datetime="2017-07-14T02:40:00.000Z"
      id="foobar"
      tabindex="0"
    >
      <span>
        2 minutes ago
      </span>
    </time>
  </strong>
</span>
`;
